### PR TITLE
Rest jobinfospec

### PR DESCRIPTION
**Pull request by dorukozturk**
This pr is for attaching jobInfoSpec to the remote jobs and for specifying a parent job on the girder end, if there is a parent specified on the celery end.

### CODE REVIEW

File: plugins/jobs/server/utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+def jobInfoSpec(job, token=None, logPrint=True):
+    """
+    Build the jobInfo specification for a task to write status and log output
+    back to a Girder job.
+
+    :param job: The job document representing the worker task.
+    :type job: dict
+    :param token: The token to use. Creates a job token if not passed.
+    :type token: str or dict
+    :param logPrint: Whether standard output from the job should be
```
Comment (cjh1) [inline plugins/jobs/server/utils.py]: show be printed?

Comment (cjh1) [reply plugins/jobs/server/utils.py]: What about stderr?

Comment (cjh1) [reply plugins/jobs/server/utils.py]: Sorry didn't see it was WIP

Comment (dorukozturk) [reply plugins/jobs/server/utils.py]: It is totally fine. Actually after talking to @kotfic , we decided to go with an event based system. I will do some refactoring.

File: plugins/jobs/server/job_rest.py [hunk 2]
```diff
@@ -81,8 +82,8 @@ def listJobs(self, userId, parentJob, types, statuses, limit, offset, sort):
         .param('title', '', required=True)
         .param('type', '', required=True)
         .modelParam('parentId', 'ID of the parent job.', model='job',
-                    plugin='jobs', level=AccessType.ADMIN,
-                    destName='parentJob', paramType='query', required=False)
+                    plugin='jobs', destName='parentJob',
+                    paramType='query', required=False, force=True)
```
Comment (cjh1) [inline plugins/jobs/server/job_rest.py]: What does this force parameter do here?

File: plugins/worker/server/__init__.py [hunk 3]
```diff
@@ -212,11 +212,33 @@ def validTransitions(event):
         event.preventDefault().addResponse(states)
 
 
+def attachParentJob(event):
+    """Attach parentJob before a model is saved."""
+    jobModel = ModelImporter.model('job', 'jobs')
+    job = event.info
+    try:
+        celeryParentTaskId = job['celeryParentTaskId']
+        parentJob = jobModel.findOne({'celeryTaskId': celeryParentTaskId})
```
Comment (cjh1) [inline plugins/worker/server/__init__.py]: We should ensure a index on celeryTaskId

**Comment by dorukozturk** [comment]
@cjh1 I changed the things you mentioned. Also added a test. PTALOneMoreTime?

**Pull request merged**